### PR TITLE
Multiple Months' worth of data in a Jupyter notebook

### PR DESCRIPTION
**Pull request by Ben-EJ**

### CODE REVIEW

File: notebooks/trends/trends-notebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+    "\n",
+    "aqi_ranges = {\n",
+    "    \"o3\": [\n",
+    "         50,\n",
+    "         100,\n",
+    "         130,\n",
+    "         240,\n",
+    "         380,\n",
+    "         800,\n",
+    "    ],\n",
```
Comment (lincent) [inline notebooks/trends/trends-notebook.ipynb:430]: I'd single line all of these arrays for space saving. Not sure you gain anything by having each element on a new line
```suggestion
    "    \"o3\": [ 50, 100, 130, 240 380 800 ],\n",
```

File: notebooks/trends/trends-notebook.ipynb [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+    "Before you run any other code cells ensure all the Python modules we need are installed. \n",
+    "\n",
+    "Do this by running the code cell below. Do so by clicking the play symbol next to the code cell on the left had side in the margin. \n",
+    "\n",
+    "Depending on what you are using to view this notebook you may have to hover over the left side of the code cell in the margin to get it to appear. This is the case in visual studio. \n",
+    "\n",
+    "If you are running this in Google Collab, you will see a round button with a play simple in it on the left hand side of the code cell, it may only appear when you hover over the code cell. \n",
+    "\n",
+    "**USEFUL TIP: If you don't want to change any settings and just want to see some graphs! On Google Collab, at the top of the screen, underneath the notebook's file name, click the “Runtime” tab on the menu bar. Then click “Run all” this will run all code cells in order.**"
+   ]
```
[the remark this answers is not in the document]
Comment (Ben-EJ) [reply notebooks/trends/trends-notebook.ipynb:60]: Probably won't hurt to keep it in though right? : )

File: notebooks/trends/trends-notebook.ipynb [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+    "\n",
+    "def update_stations_to_display(stations_input):\n",
+    "      global stations_to_display\n",
+    "      stations_to_display = stations_input\n",
+    "\n",
+    "for pollutant in in_situ_processed_data:\n",
+    "    for measuring_station, station_data in in_situ_processed_data[pollutant].items():\n",
+    "        stations_with_duplicates.append(measuring_station)\n",
+    "\n",
+    "stations = list(set(stations_with_duplicates))\n",
```
[the remark this answers is not in the document]
Comment (lincent) [reply notebooks/trends/trends-notebook.ipynb:384]: can a set not do what you need? I would still eliminate the original list, and add them directly to the set. If you definiteely need a list then do the conversion then, but no point doing list to set to list

File: notebooks/trends/trends-notebook.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (lincent) [reply notebooks/trends/trends-notebook.ipynb:60]: I don't really mind, I just see it as a bit unnecessary and more stuff to read

File: notebooks/trends/trends-notebook.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ben-EJ) [reply notebooks/trends/trends-notebook.ipynb:384]: No, you can't index into a set like you can with a list. 
image

File: notebooks/trends/trends-notebook.ipynb [hunk 4]
```diff
@@ -0,0 +1,518 @@
...
+    "interact(update_time_period, months_input=time_period_widget)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Make API Request\n",
+    "\n",
+    "Run the below code cell to get the data from the API. The top code cell will get the in_situ data and the one below that will get the forecast data.\n",
```
Comment (mwalker-scottlogic) [inline notebooks/trends/trends-notebook.ipynb:183]:
```suggestion
    "Run the below code cell to get the data from the API. The first code cell will get the in_situ data and the second will get the forecast data.\n",
```

File: notebooks/trends/trends-notebook.ipynb [hunk 5]
```diff
@@ -0,0 +1,518 @@
...
+    "    current_time = current_time + timedelta(days= 1)\n",
+    "\n",
+    "print(forecast_response.status_code)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Prepare Data to display\n",
```
Comment (mwalker-scottlogic) [inline notebooks/trends/trends-notebook.ipynb:270]:
```suggestion
    "# Prepare Data to Display\n",
```

File: notebooks/trends/trends-notebook.ipynb [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+    "\n",
+    "The below code will prepare data to display on the graph. It groups the data by pollutant and within that by measuring station. The top code cell will process the in situ data and the one below that will process the forecast data."
```
Comment (mwalker-scottlogic) [inline notebooks/trends/trends-notebook.ipynb:272]:
```suggestion
    "The below code will prepare data to display on the graph. It groups the data by pollutant and within that by measuring station. The first code cell will process the in situ data and the second will process the forecast data."
```

File: notebooks/trends/trends-notebook.ipynb [hunk 7]
```diff
@@ -0,0 +1,518 @@
...
+    "    if \"so2\" in forecast: \n",
+    "        forecast_processed_data[\"so2\"][\"values\"].append(forecast[\"so2\"][\"value\"])\n",
+    "        forecast_processed_data[\"so2\"][\"times\"].append(forecast_date)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Select Measuring stations to display\n",
```
Comment (mwalker-scottlogic) [inline notebooks/trends/trends-notebook.ipynb:359]:
```suggestion
    "# Select Measuring Stations to Display\n",
```

File: notebooks/trends/trends-notebook.ipynb [hunk 8]
```diff
@@ -0,0 +1,518 @@
...
+    "\n",
+    "The list of stations is derived from the data obtained from the API call. \n",
+    "\n",
+    "**ATTENTION: Some of the stations may not have data for a given pollutant. Therefore if you deselect all stations that have data for a given pollutant that pollutants graph will no longer render.**"
```
Comment (mwalker-scottlogic) [inline notebooks/trends/trends-notebook.ipynb:363]:
```suggestion
    "**ATTENTION: Some of the stations may not have data for a given pollutant. Therefore if you deselect all stations that have data for a given pollutant, that pollutants graph will no longer render.**"
```

**Review by sebsteinig — APPROVED** [review]

**Pull request merged**